feat: allow batching of simulatetx

Thunkar opened the pull request:
Extends the `.batch` method of the wallet interface to support `.simulateTx`, since there's no good reason not to and it enables more meaningful interactions.

Also tweaked `BatchCall` to use it, so the feature is immediately available to the ecosystem...but again, it's probably going to be dead soon.

benesjan commented on yarn-project/aztec.js/src/contract/batch_call.ts:
it says it simulates a tx but then it says it supports utility functions which cannot be included in a tx. Please fix this lie here.

Thunkar replied:
Fixed!

benesjan reviewed (approved):
LGTM but please fix the docs of simulate fn of batch_call

Thunkar committed:
feat: allow batching of simulatetx

Extends the `.batch` method of the wallet interface to support `.simulateTx`, since there's no good reason not to and it enables more meaningful interactions.

Also tweaked `BatchCall` to use it, so the feature is immediately available to the ecosystem...but again, it's probably going to be dead soon.
